src/components/JujuCLI/JujuCLI.tsx

```diff
@@ -106,7 +112,28 @@ const JujuCLI = () => {
       .replace("ws://", "")
       .replace("wss://", "")
       .replace("/api", "") || null;
-  const wsProtocol = primaryControllerData?.[0].split("://")[0];
+  const wsProtocol = primaryControllerData?.[0].split("://")[0] ?? "wss";
+
+  const wsAddress = useMemo(() => {
+    if (!controllerWSHost || !modelUUID) {
+      return null;
+    }
+    return `${wsProtocol}://${controllerWSHost}/model/${modelUUID}/commands`;
+  }, [controllerWSHost, modelUUID, wsProtocol]);
+
+  const originalWSOrigin = wsAddress
+    ? `${new URL(wsAddress).origin}/api`
+    : null;
+  const activeUser = useAppSelector((state) =>
+    getActiveUserTag(state, originalWSOrigin),
```

Comment:
Looking at the logic that was originally implemented, it seems that it does a lot of logic to transform the websocket address a number of times, and the end result is exactly what we had to begin with, which is `primaryControllerData?.[0]` so I think this can just be `getActiveUserTag(state, primaryControllerData?.[0])` and then we don't need the `wsAddress` logic in this component.